src/arg-counter.c

```diff
@@ -61,7 +56,6 @@ void counters_shift(struct counters* p_counters) {
   // Swap the counters data
   SWAP(struct vctrs_arg, p_counters->curr_counter, p_counters->next_counter);
   SWAP(r_ssize*, p_counters->curr_counter_arg_data.i, p_counters->next_counter_arg_data.i);
-  SWAP(r_ssize*, p_counters->curr_counter_arg_data.names_i, p_counters->next_counter_arg_data.names_i);
 
   // Update the handles to `vctrs_arg`
   p_counters->curr_arg = (struct vctrs_arg*) &p_counters->curr_counter;
```

Comment:
The first actual issue was the the line that is just below this, which is:

```
  // Update the current index
  p_counters->curr = p_counters->next;
```

was not also doing 

```
  p_counters->names_curr = p_counters->names_next;
```

so when we would index into the `names` at error reporting time, we'd have the wrong index!

I've fixed that by realizing that we don't need both `r_ssize curr/next` and `r_ssize names_curr/names_next`. They are tracking exactly the same thing, so we now just use `r_ssize curr/next` for everything. That immediately fixes the issue.